lucene/core/src/java/org/apache/lucene/util/automaton/SortedIntSet.java

```diff
@@ -77,6 +88,7 @@ public void incr(int num) {
         values[i] = num;
         counts[i] = 1;
         upto++;
+        stale = true;
```

Comment:
How come the code didn't need/use `stale` before?

Reply:
Because it didn't track whether the cached hash code was stale and relied on the caller to do it. I feel like it is more clear to manage it internally.

Reply:
oh, found the replacement we should use: ArrayUtil.copyOfSubArray